test/Call.test.mo

```diff
@@ -0,0 +1,107 @@
+import Blob "mo:base/Blob";
+import { suite; test; expect } "mo:test/async";
+
+import IC "../src/lib";
+import Call "../src/Call";
+
+actor {
+  public func runTests() : async () {
+    await test(
+      "createCanister should succeed",
+      func() : async () {
+        ignore await Call.createCanister(createCanisterArgs);
+      },
+    );
+
+    await test(
+      "createCanister cost should be exact",
+      func() : async () {
+        let cycles = Call.Cost.createCanister();
+        ignore await (with cycles) IC.ic.create_canister(createCanisterArgs);
+        await expectThrows(
+          func() : async () {
+            ignore await (with cycles = cycles - 1) IC.ic.create_canister(createCanisterArgs);
+          }
+        );
+      },
+    );
+
+    await test(
+      "httpRequest should succeed",
+      func() : async () {
+        ignore await Call.httpRequest(request);
+        ignore await Call.httpRequest({ request with headers });
+        ignore await Call.httpRequest({ request with body });
+        ignore await Call.httpRequest({ request with max_response_bytes });
+        ignore await Call.httpRequest({
+          request with headers;
+          body;
+          max_response_bytes;
+        });
+        ignore await Call.httpRequest({ request with transform });
+      },
+    );
+
+    await suite(
+      "httpRequest cost should be exact",
+      func() : async () {
+        await test("default", httpRequestExactCost(request));
+        await test("with headers", httpRequestExactCost({ request with headers }));
+        await test("with body", httpRequestExactCost({ request with body }));
+        await test("with max_response_bytes", httpRequestExactCost({ request with max_response_bytes }));
+        await test("with all above", httpRequestExactCost({ request with headers; body; max_response_bytes }));
+
+        // Future work: transform can't be exact yet
+        // await test("with transform", httpRequestExactCost({ request with transform }));
+      },
+    );
+  };
+
+  func httpRequestExactCost(request : IC.HttpRequestArgs) : () -> async () = func() : async () {
+    let cycles = Call.Cost.httpRequest(request);
+    ignore await (with cycles) IC.ic.http_request(request);
+    await expectThrows(
+      func() : async () {
+        ignore await (with cycles = cycles - 1) IC.ic.http_request(request);
+      }
+    );
```

Comment:
Perhaps `await expect.call(...).reject();` would work here? But there is a check for `#canister_reject` error code, not sure why I added it 🤔

https://github.com/ZenVoich/test?tab=readme-ov-file#expectcall
https://mops.one/test/code/src/expect/expect-call.mo#ExpectCall.reject